presto-main/src/main/java/com/facebook/presto/operator/PartitionedLookupSourceFactory.java

```diff
@@ -82,6 +78,9 @@
     @GuardedBy("this")
     private final List<SettableFuture<LookupSource>> lookupSourceFutures = new ArrayList<>();
 
+    @GuardedBy("this")
+    private final Map<Integer, CompletableFuture<LookupSource>> unspillLookupSourceFuturesByPartitionCache = new HashMap<>();
```

Comment:
I don't think we need this map here, since it always should contain at most one element. Having two fields: integer and one `Optional<CompletableFuture<LookupSource>>`  would be more implicit.